src/nvim/option.c

```diff
@@ -3738,6 +3738,10 @@ did_set_string_option (
       if (varp == &p_enc && *curbuf->b_p_keymap != NUL)
         (void)keymap_init();
 
+      if (varp == &p_enc) {
+        ui_update_encoding();
```

Comment:
Changing `'encoding'` after initialization is not meaningfully supported by even Vim:

    NOTE: Changing this option will not change the encoding of the
    existing text in Vim.  It may cause non-ASCII text to become invalid.
    It should normally be kept at its default value, or set when Vim
    starts up.  See |multibyte|.  To reload the menus see |:menutrans|.

so I don't think we really need to support this. Can we just check `enc_utf8` in `term_input_new()`?